aiocache/backends/redis.py

```diff
@@ -52,163 +39,171 @@ def __init__(
         )
-        self.__pool_lock = None
-        self._loop = loop
-        self._pool = None
-
-    @property
-    def _pool_lock(self):
-        if self.__pool_lock is None:
-            self.__pool_lock = asyncio.Lock()
-        return self.__pool_lock
-
-    async def acquire_conn(self):
-        await self._get_pool()
-        conn = await self._pool.acquire()
-        conn = aioredis.Redis(conn)
-        return conn
-
-    async def release_conn(self, _conn):
-        self._pool.release(_conn)
-
-    @conn
+
+        redis_kwargs = {
+            "host": self.endpoint,
+            "port": self.port,
+            "db": self.db,
+            "password": self.password,
+            # NOTE: decoding can't be controlled on API level after switching to
+            #  redis, we need to disable decoding on global/connection level.
+            #  Cause some of the values are save as bytes directly, like pickle
+            #  serialized values, which may raise exc when decoded with 'utf-8'.
+            "decode_responses": False,
+            # parameter names are different
+            "socket_connect_timeout": self.create_connection_timeout,
+            "max_connections": self.pool_max_size,
+        }
+        self.client = redis.Redis(**redis_kwargs)
+
     async def _get(self, key, encoding="utf-8", _conn=None):
-        return await _conn.get(key, encoding=encoding)
+        value = await self.client.get(key)
+        if encoding is None or value is None:
+            return value
+        return value.decode(encoding)
 
-    @conn
     async def _gets(self, key, encoding="utf-8", _conn=None):
         return await self._get(key, encoding=encoding, _conn=_conn)
 
-    @conn
     async def _multi_get(self, keys, encoding="utf-8", _conn=None):
-        return await _conn.mget(*keys, encoding=encoding)
+        values = []
+        for value in await self.client.mget(*keys):
+            if encoding is None or value is None:
+                values.append(value)
+            else:
+                values.append(value.decode(encoding))
+        return values
 
-    @conn
     async def _set(self, key, value, ttl=None, _cas_token=None, _conn=None):
         if _cas_token is not None:
             return await self._cas(key, value, _cas_token, ttl=ttl, _conn=_conn)
         if ttl is None:
-            return await _conn.set(key, value)
-        return await _conn.setex(key, ttl, value)
+            return await self.client.set(key, value)
+        if isinstance(ttl, float):
+            ttl = int(ttl * 1000)
+            return await self.client.psetex(key, ttl, value)
+        return await self.client.setex(key, ttl, value)
 
-    @conn
     async def _cas(self, key, value, token, ttl=None, _conn=None):
         args = [value, token]
         if ttl is not None:
             if isinstance(ttl, float):
                 args += ["PX", int(ttl * 1000)]
             else:
                 args += ["EX", ttl]
```

Comment:
On an unrelated point, these should probably be `args.extend(...)`, rather than doing concatenation.